test-data/unit/lib-stub/mypy_extensions.pyi

```diff
@@ -1,8 +1,10 @@
-from typing import Dict, Type, TypeVar
+from typing import Dict, Type, TypeVar, Callable
 
 T = TypeVar('T')
 
 
 def TypedDict(typename: str, fields: Dict[str, Type[T]]) -> Type[dict]: pass
 
 class NoReturn: pass
+
+def decorated_type(t: Type) -> Callable[[T], T]: pass
```

Comment:
The argument type should be `Any`, since some valid types aren't type objects (e.g. `None`).